File: swagger-example/spring-swagger/src/main/java/com/example/demo/model/UserModel.java
package com.example.demo.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * UserModel
 * 
 * @author zww19
 * @since 2022年1月29日,下午8:25:09
 */
@ApiModel(description = "用户模型")
@Getter
@Setter
@ToString
public class UserModel {
	@ApiModelProperty(value = "用户ID", required = true)
	@NotNull(message = "用户ID不能为空")
	@Positive(message = "用户ID必须大于0")
	private Integer userId;
	@ApiModelProperty(value = "用户名", required = true)
	@NotBlank(message = "用户名不能为空")
	private String userName;

}

File: angular-example/my-angular-v2/src/app/shared/user/user.model.ts
/**
 * @description 用户信息数据模型
 * @author zww
 */
export class User {
  /**
   * @description 用户ID
   */
  userId: number | string;
  /**
   * @description 用户名称
   */
  userName: string;

  /**
   * @description 创建用户
   * @param userId 用户ID
   * @param userName 用户名称
   */
  constructor(userId: number | string, userName: string) {
    this.userId = userId;
    this.userName = userName;
  }
}
